addons/traefik/1.7.x/traefik-11.yaml

```diff
@@ -0,0 +1,130 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: traefik
+  labels:
+    kubeaddons.mesosphere.io/name: traefik
+    kubeaddons.mesosphere.io/provides: ingresscontroller
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "1.7.23-11"
+    appversion.kubeaddons.mesosphere.io/traefik: "1.7.23"
+    endpoint.kubeaddons.mesosphere.io/traefik: "/ops/portal/traefik"
+    docs.kubeaddons.mesosphere.io/traefik: "https://docs.traefik.io/v1.7"
+    values.chart.helm.kubeaddons.mesosphere.io/traefik: "https://raw.githubusercontent.com/mesosphere/charts/00b019ef3610ca8221a8cf283b4d7046a50702c4/staging/traefik/values.yaml"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.6
```

Comment:
well if a tool doesnt use them for earlier versions, does this harm the deployment?

Reply:
I haven't tested this on k8s < 1.17, but the k8s docs have a vague warning against using a `topologyKey` that isn't present in all node labels: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity

> Pod anti-affinity requires nodes to be consistently labelled, in other words every node in the cluster must have an appropriate label matching topologyKey. If some or all nodes are missing the specified topologyKey label, it can lead to unintended behavior.

So I guess we should avoid that.

I think the simplest way to resolve this without making changes to supported k8s versions is to use the deprecated zone label, which is still present in 1.17. https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#failure-domainbetakubernetesiozone

Reply:
I went ahead and gave the pods `kubeaddons.mesosphere.io/name` labels in order to avoid inventing my own label: https://github.com/mesosphere/kubernetes-base-addons/pull/292/commits/be789c5163a47942b356fdaefad7775a173f94e9

Reply:
https://github.com/mesosphere/kubernetes-base-addons/pull/292/commits/392e19e7c5215ea203fcfed551362450be7f8873

Reply:
My only concern with going _backward_ to (now deprecated) failure-domain labels is the fact that:
* Kubernetes 1.17 is now 2 months GA
* Kubernetes 1.19 will be released in August, at which time 1.16 support basically ends.

As such, *after August* I don't think there will be any supported versions of Kubernetes which _do not_ support `topology.kubernetes.io/zone`. 

Why aren't we testing against multiple Kubernetes versions?

Isn't the design of `Addon.spec.kubernetes.minSupportedVersion` specifically to allow selection of compatible addons?

Why would we prefer to _fall back_ instead of _lean forward_?

cc @joejulian

Reply:
I don't think support for that ever made it into catalog and nothing but kommander even uses catalog. We could make that part of 1.7 and upgrade from the deprecated configuration at that time but today I think this is the only option.

Reply:
Also:
- we haven't been keeping up with k8s releases
- we need to support eks which is always behind k8s support
- I believe you said that 1.16 is going to be the lts version, so we would need to maintain support for that anyway.

Reply:
OK, I think for the immediate objectives, the existing (deprecated) label is viable. 

The particular versions that **we** will support with LTS is undecided at this time, however it will be *at earliest* Kubernetes 1.16. Customer contract details dictate that it will be decided _at the time the customer pays for LTS support_ based on **one of the versions we support**. It's possible that, under some future conditions, we'd offer extended support for Kubernetes that are already beyond their community support timeframe (i.e. EOL by community).

Community Kubernetes is working to establish an N-3 support policy (extending from the previous N-2) taking effect with Kubernetes 1.19, which would include 1.16 -- effectively extending community support to roughly 1 year (depending on release cadence). 

In the near future, we need to:
* Start testing all supported Kubernetes on associated KBA branches. 
* Start testing *newer* (unsupported) Kubernetes in unstable KBA branches, so we can know *much earlier* where there will be problems.
* Start using the Catalog functionality in kubeaddons, as appropriate, to select compatible addon revisions automatically. 
* Start versioning addons to support appropriate capabilities for various Kubernetes releases.

Relatedly, we can probably *stop* supporting things for 1.15 when the KBA release tag says `stable-1.17-...` or some variant thereof. Probably we should start pruning the revisions with each release.